analyzers/src/SonarAnalyzer.VisualBasic/Rules/ExpectedExceptionAttributeShouldNotBeUsed.cs

```diff
@@ -18,15 +18,44 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
-namespace SonarAnalyzer.Rules.VisualBasic
+namespace SonarAnalyzer.Rules.VisualBasic;
+
+[DiagnosticAnalyzer(LanguageNames.VisualBasic)]
+public sealed class ExpectedExceptionAttributeShouldNotBeUsed : ExpectedExceptionAttributeShouldNotBeUsedBase<SyntaxKind>
 {
-    [DiagnosticAnalyzer(LanguageNames.VisualBasic)]
-    public sealed class ExpectedExceptionAttributeShouldNotBeUsed : ExpectedExceptionAttributeShouldNotBeUsedBase<SyntaxKind>
+    protected override ILanguageFacade<SyntaxKind> Language => VisualBasicFacade.Instance;
+
+    protected override bool HasMultiLineBody(SyntaxNode node) =>
+        node.Parent is MethodBlockSyntax { Statements.Count: > 1 };
+
+    protected override bool AssertInCatchFinallyBlock(SyntaxNode node)
+    {
+        var walker = new CatchFinallyAssertion();
+        foreach (var x in node.Parent.DescendantNodes(_ => true).Where(x => x.IsAnyKind(SyntaxKind.CatchBlock, SyntaxKind.FinallyBlock)))
+        {
+            walker.SafeVisit(x);
```

Comment:
```suggestion
            if (!walker.HasAssertion) 
            {
               walker.SafeVisit(x);
            }
```

1. Performance improvement. No need to look for other blocks if we've already found an assertion method.
2. It fixes an FP when `catch` block has an assertion method, but the `finally` block or another `catch` block that follows it doesn't. See [this FP](https://github.com/SonarSource/sonar-dotnet/blob/46fba4e2f32374a5409bb2d02ba2f05a9e537a8e/analyzers/tests/SonarAnalyzer.Test/TestCases/ExpectedExceptionAttributeShouldNotBeUsed.MsTest.cs#L158).

Reply:
I meant to fix this FP and forgot about it 🤦 
Thanks!